pkg/driver/node.go

```diff
@@ -848,29 +839,104 @@ func collectMountOptions(fsType string, mntFlags []string) []string {
 	return options
 }
 
+// startNotReadyTaintWatcher launches a short‑lived Node informer that removes the
+// ebs.csi.aws.com/agent‑not‑ready taint as soon as both the taint is present and
+// the driver’s Allocatable value is set. The watcher stops automatically after
+// maxWatchDuration or immediately after successful removal.
+func startNotReadyTaintWatcher(clientset kubernetes.Interface, maxWatchDuration time.Duration) {
+	nodeName := os.Getenv("CSI_NODE_NAME")
+	if nodeName == "" {
+		klog.V(4).InfoS("CSI_NODE_NAME missing, skipping taint watcher")
+		return
+	}
+
+	stopCh := make(chan struct{})
+	var closeOnce sync.Once
+	closeWatch := func() { closeOnce.Do(func() { close(stopCh) }) }
+
+	factory := informers.NewSharedInformerFactoryWithOptions(
+		clientset,
+		0,
```

Comment:
```suggestion
		0, // No resync
```

I had to look up what the `0` does here, maybe add a comment like this to indicate?

Reply:
Done ✅